### PR TITLE
Rename all SSO references to SS4O

### LINKED ISSUE #1448: [BUG] replace SSO with SS4O initials for Simple Schema
**What is the bug?**
replace all SSO references with SS4O initials for Simple Schema 

**How can one reproduce the bug?**
N/A

**What is the expected behavior?**
N/A

**What is your host/environment?**
N/A

**Do you have any screenshots?**
N/A

**Do you have any additional context?**
Add any other context about the problem.

### PR DESCRIPTION
**Pull request by Swiddis**
### Description

PR does what it says on the tin, per #1448 

The main substitution (minus one file rename) was done with the following shell command, it may be better to review this for bugs than the individual diffs:

```sh
grep -rli "[^a-z]sso[^a-z]" . --exclude-dir=.git | xargs sed -i -r 's/([^a-z]ss)(o[^a-z])/\14\2/i'
```

### Issues Resolved

Resolves #1448 

### Check List

- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integrations/nginx/assets/display/ss4o-logs-dashboard-new.ndjson
Comment (YANG-DB) [inline integrations/nginx/assets/display/ss4o-logs-dashboard-new.ndjson:1]: why this file was effected ? it doesn’t contain sso ...

**Review by YANG-DB — CHANGES_REQUESTED** [review]

File: integrations/nginx/assets/display/ss4o-logs-dashboard-new.ndjson
Comment (Swiddis) [reply integrations/nginx/assets/display/ss4o-logs-dashboard-new.ndjson:1]: It's in the `title`, column `8949`. `"title":"sso_logs-*-*"`

File: integrations/nginx/info/nginx.png
[the remark this answers is not in the document]
Comment (Swiddis) [reply integrations/nginx/info/nginx.png:1]: Yep, that happens if you accidentally do a find-replace for a bytestring in a png.

For any future readers, you can use the `-I` flag in grep to tell it to skip binary files.

File: docs/Integration-API.md [hunk 1]
```diff
@@ -199,7 +199,7 @@ Once user changes the data-stream / index pattern - this will be reflected in ev
  - update the asset name (according to the `instance_name` field)
    - `${instance_name}-assetName.json`, this can also be extended using more configurable patterns such as `${instance_name}-{dataset}-{namespace}-assetName.json`
  - update the index template's `index_pattern` field with the added pattern 
-   - "index_patterns":` ["sso_logs-*-*"]` -> `["sso_logs-*-*", "myLogs-*"]`
+   - "index_patterns":` ["ss4o_logs-*-*"]` -> `["sso_logs-*-*", "myLogs-*"]`
```
Comment (derek-ho) [inline docs/Integration-API.md:202]: miss one here?

File: integrations/nginx/info/fluent-bit/fluentbit.png
Comment (derek-ho) [inline integrations/nginx/info/fluent-bit/fluentbit.png:1]: why do these files need to be re-named? Am I missing something

Comment (Swiddis) [reply integrations/nginx/info/fluent-bit/fluentbit.png:1]: Just correcting a misspelled directory name, technically not in the scope of the PR but seems harmless enough to just add it in. I can make it a separate PR if you prefer?

File: src/main/kotlin/org/opensearch/observability/index/ObservabilityTracesIndex.kt [hunk 2]
```diff
@@ -26,9 +26,9 @@ import java.util.*
  */
 internal object ObservabilityTracesIndex : LifecycleListener() {
     private val log by logger(ObservabilityTracesIndex::class.java)
-    private const val TRACES_MAPPING_TEMPLATE_NAME = "sso_trace_template"
+    private const val TRACES_MAPPING_TEMPLATE_NAME = "ss4o_trace_template"
```
[the remark this answers is not in the document]
Comment (Swiddis) [reply src/main/kotlin/org/opensearch/observability/index/ObservabilityTracesIndex.kt:29]: I think one option is just add a deprecation warning. When integration plugin starts up, check for sso indices, and if present log a warning like `Detected indices with 'sso_*' pattern, which is deprecated. The new 'ss4o_*' pattern will be created and used by default`.

Even better would be making it configurable, which could be implemented at import-time. The import method could involve a step for checking for a configured index name, and if one is found, replace the integration's index patterns in-place.

[the remark this answers is not in the document]
Comment (YANG-DB) [reply src/main/kotlin/org/opensearch/observability/index/ObservabilityTracesIndex.kt:29]: agree

**Review by YANG-DB — DISMISSED** [review]

**Review by derek-ho — APPROVED** [review]

**Pull request merged**